### PR TITLE
Oscar - Eliminates Backend Files for Restaurant, UCSBDiningCommons, UCSBDates (New PR from main)

### LINKED ISSUE #19: Eliminate Restaurant/UCSBDates Backend

### PR DESCRIPTION
**Pull request by obenedek20**
Closes #19 

Dokku: https://rec-obenedek20-dev.dokku-08.cs.ucsb.edu/

You should see that all references/api routes to restaurants, UCSBDates, and ucsbdiningcommons are removed
This is a new version of the PR from earlier that is now closed (PR #41) that is branched straight from main and does not rely on PR #39

### CODE REVIEW

**Review by divyanipunj — DISMISSED** [review]
LGTM - the new files seem to properly implement what the professor requested.

File: src/main/java/edu/ucsb/cs156/rec/controllers/DummyController.java [hunk 1]
```diff
@@ -0,0 +1,26 @@
+package edu.ucsb.cs156.rec.controllers;
```
Comment (pconrad) [inline src/main/java/edu/ucsb/cs156/rec/controllers/DummyController.java]: Please move `src/main/java/edu/ucsb/cs156/rec/controllers/DummyController.java` to 
`src/test/java/edu/ucsb/cs156/rec/controllers/DummyController.java`.

It is not a real controller, and exists only to help with test coverage.  So it should not be under `/src/main/java` but under `/src/test/java`.  That way it doesn't really show in the real app; only when testing.

**Review by pconrad — CHANGES_REQUESTED** [review]
Please move the DummyController to the test hierarchy.

File: src/main/java/edu/ucsb/cs156/rec/controllers/DummyController.java (hunk 1 above)
Comment (pconrad) [reply src/main/java/edu/ucsb/cs156/rec/controllers/DummyController.java]: See: https://ucsb-cs156-f24.slack.com/archives/C080JHJN9QE/p1732427414430999

**Review by pconrad — APPROVED** [review]
LGTM!

**Pull request merged**